src/docfx/build/xref/XrefResolver.cs

```diff
@@ -137,6 +137,13 @@ private string RemoveSharingHost(string url, string hostName)
                 return url.Substring(hostName.Length);
             }
 
+            // TODO: this workaround can be removed when all xref related repos migrated to v3
+            if (hostName.Equals("https://docs.microsoft.com", StringComparison.OrdinalIgnoreCase)
+                        && url.StartsWith($"https://review.docs.microsoft.com/", StringComparison.OrdinalIgnoreCase))
+            {
+                return url.Substring("https://review.docs.microsoft.com".Length);
```

Comment:
when url starts with `https://docs.microsoft.com` this substring gives wrong result.

Reply:
Do you mean url starts with `https://docs.microsoft.com` and host name is also `https://docs.microsoft.com`?